ktor-test-dispatcher/mingwX64/src/TestMingwX64.kt

```diff
@@ -1,12 +1,53 @@
 package io.ktor.test.dispatcher
 
 import kotlinx.coroutines.*
+import platform.posix.*
 import kotlin.coroutines.*
+import kotlin.native.concurrent.*
+import kotlin.system.*
+import kotlin.time.Duration.Companion.milliseconds
+
+private var TEST_WORKER: Worker = createTestWorker()
+private val SLEEP_TIME: UInt = 10.milliseconds.inWholeMicroseconds.toUInt()
 
 /**
  * Test runner for native suspend tests.
  */
 public actual fun testSuspend(
     context: CoroutineContext,
+    timeout: Long,
     block: suspend CoroutineScope.() -> Unit
-): Unit = runBlocking(context, block)
+) {
+    executeInWorker(timeout) {
+        runBlocking {
+            block()
+        }
+    }
+}
+
+private fun executeInWorker(timeout: Long, block: () -> Unit) {
```

Comment:
win, mac and linux workers are similar. can it be shared?

Reply:
I can try moving this to posix

Reply:
Done